src/main/java/org/apache/sysds/lops/PartialAggregate.java

```diff
@@ -342,11 +342,20 @@ else if( dir == Direction.Col )
 			}
 
 			case COUNT_DISTINCT: {
-				if(dir == Direction.RowCol )
-					return "uacd";
+				switch (dir) {
+					case RowCol: return "uacd";
+					case Row: return "uacdr";
+					case Col: return "uacdc";
```

Comment:
Maybe throw an exception here in the default case and not use the break.
This would make errors more informative.